### DISCUSSION TITLE
Extension activation order and status

### DISCUSSION THREAD

**Discussion by daniel-a126** [microsoft/vscode-discussions#1241] [Extension Development QnA] [answered]
We have learned the hard way that the order of actions while _activating_ the extension matters. For example, you may loose some commands if you first subscribe views and then commands, while doing in reverse eventually delivers those commands to the view after the extension gets activated (not a great CX though, because the user _clicks_ and _nothing_ seems to be happening...

Anyway... Is there any guidance on the order of the activation actions for a frictionless activation?

Also, we were not able to find a way to tell if the extension is still activating or not (apart from setting our own _context variable_. Is there a way or should jump the variable hoop?

Thank you!

**Comment by tamuratak** [comment, marked as answer]
> setting our own context variable

I think it is an intended solution. For example, GitLens heavily uses context variables:

- https://github.com/search?q=repo%3Agitkraken%2Fvscode-gitlens%20setcontext&type=code
- https://github.com/gitkraken/vscode-gitlens/blob/da47858caa97552f3ec350342f74bb1bd1c1964d/package.json#L17196

A related issue: microsoft/vscode/issues/10471

**Reply by daniel-a126** [reply]
Thanks!
